internal/services/signalr/signalr_service_resource.go

```diff
@@ -310,6 +359,18 @@ func resourceArmSignalRServiceRead(d *pluginsdk.ResourceData, meta interface{})
 				return fmt.Errorf("setting `features`: %+v", err)
 			}
 
+			for _, feature := range *props.Features {
+				if feature.Flag == signalr.FeatureFlagsEnableConnectivityLogs {
+					d.Set("connectivity_logs_enabled", feature.Value)
+				}
+				if feature.Flag == signalr.FeatureFlagsEnableMessagingLogs {
+					d.Set("messaging_logs_enabled", feature.Value)
+				}
+				if feature.Flag == signalr.FeatureFlagsServiceMode {
+					d.Set("service_mode", feature.Value)
+				}
+			}
```

Comment:
these fields always need to be set into the state, so can we make this:

```suggestion
			connectivityLogsEnabled := false
			messagingLogsEnabled := false
			serviceMode := "Default"
			for _, feature := range *props.Features {
				if feature.Flag == signalr.FeatureFlagsEnableConnectivityLogs {
					connectivityLogsEnabled = strings.EqualsFold(feature.Value, "True")
				}
				if feature.Flag == signalr.FeatureFlagsEnableMessagingLogs {
					messagingLogsEnabled = strings.EqualsFold(feature.Value, "True")
				}
				if feature.Flag == signalr.FeatureFlagsServiceMode {
					serviceMode = feature.Value
				}
			}
			d.Set("connectivity_logs_enabled", connectivityLogsEnabled)
			d.Set("messaging_logs_enabled", messagingLogsEnabled)
			d.Set("service_mode", serviceMode)
```